config/routes.rb

```diff
@@ -16,7 +16,9 @@
   end
 
   namespace :api do
-    resources :jobs, only: %i[index show], controller: 'vacancies'
+    scope 'v:api_version', api_version: /[1]/ do
```

Comment:
If we needed to bump to version 2, is the plan that we would add a new block with a different controller name like this:

```
scope 'v:api_version', api_version: /[2]/ do
  resources :jobs, only: %i[index show], controller: 'v2/vacancies'
end
```

If so, might it be prudent to namespace the existing controller to v1?

Reply:
Yes. We would also have to edit v1. The following example should do it.

```
  namespace :api do
    scope 'v:api_version', api_version: /[1]/ do
      resources :jobs, only: %i[index show], controller: 'v1/vacancies'
    end

    scope 'v:api_version', api_version: /[2]/ do
      resources :jobs, only: %i[index show], controller: 'v2/vacancies'
    end

    get '*path', :to => redirect("/api/v2/%{path}")
  end
```

Reply:
The wildcard redirector worries me a bit. 

Whilst I can see the utility in helping users who haven't declared a version number and want to get started quickly with the current latest version, I can also see that by being relaxed it's omission we may accidentally break automated integrations with us. When we make breaking changes in v2 all requests set up initially with `api/jobs` as the path will suddenly stop working or behalf unexpectedly. I think this is one of the benefits of versioning the API. I would think about returning a 404 instead, and ensure all requests specify a version number explicitly.

Do you see this being an issue?

Reply:
I'm also unclear about your thoughts on changing the controller's namespace now, rather than later when/if v2 comes?

Reply:
I don't think we should change it now. There is no plan to extend this API to be writable at any point so I don't see the value in making a change like that.

Reply:
We are more likely to add or remove data from the representation of job listings. eg. flexible working. The controller name spacing can be done later when we need to do it 👍 

The wildcard redirector is still a worry for me. Thoughts?

Reply:
Regarding the redirection, I came across it when looking up how to best structure the routes for the API and it seems to be common practise. There is a possibility that it will break, however, the goal is to enable the users to be able to discover the latest version rather than to use it to retrieve data by default. I'm happy to take it out though.

Reply:
@tahb removed

Reply:
Was there a particular article you found that you could link?

Reply:
https://stackoverflow.com/questions/389169/best-practices-for-api-versioning
https://stackoverflow.com/questions/9627546/api-versioning-for-rails-routes

Reply:
Thanks for the links.

I'm unconvinced by their rationale and am struggling to see the valuable benefits.  

I feel like the safer route is to require the API user to explicitly choose a version which might take them a few more minutes to pick or discover however it should give long term assurance of a stable interface for integration.

A luxury of this route is that we can always add an alias in later if we really need to. We cannot however responsibly remove it once we put it in users hands.

Reply:
@tahb I have already removed that commit.